File: src/main/java/org/example/chessplatformbe/controller/DTO/Response/SpectatorResponseDTO.java
package org.example.chessplatformbe.controller.DTO.Response;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.Getter;

@Data
@EqualsAndHashCode(callSuper = true)
@Getter
public class SpectatorResponseDTO extends UserResponseDTO  {

    private int playerElo;

    private boolean isChatBanned;

    private boolean isGameBanned;

    private int noOfGamesPlayed;

    private String chroma;

    private boolean hasPass;
}

File: src/controller/dto/response/spectator_response_dto.rs
pub struct SpectatorResponseDTO {
    id: u32,
    age: u8,
    username: String,
    display_name: String,
    nationality: String,

    player_elo: u32,
    is_chat_banned: bool,
    is_game_banned: bool,
    no_of_games_played: u32,
    chroma: String,
    has_pass: bool,
}

impl SpectatorResponseDTO {
    pub fn new(
        id: u32,
        age: u8,
        username: String,
        display_name: String,
        nationality: String,
        player_elo: u32,
        is_chat_banned: bool,
        is_game_banned: bool,
        no_of_games_played: u32,
        chroma: String,
        has_pass: bool,
    ) -> Self {
        Self {
            id,
            age,
            username,
            display_name,
            nationality,
            player_elo,
            is_chat_banned,
            is_game_banned,
            no_of_games_played,
            chroma,
            has_pass,
        }
    }
}
